core/src/dird/ua_select.cc

```diff
@@ -1952,28 +1953,38 @@ bool GetUserJobTypeListSelection(UaContext* ua,
   return true;
 }
 
-bool GetUserJobStatusSelection(UaContext* ua, int* jobstatus)
+bool GetUserJobStatusSelection(UaContext* ua, std::vector<char>& jobstatuslist)
 {
-  int i;
-
-  if ((i = FindArgWithValue(ua, NT_("jobstatus"))) >= 0) {
-    if (strlen(ua->argv[i]) == 1 && ua->argv[i][0] >= 'A'
-        && ua->argv[i][0] <= 'z') {
-      *jobstatus = ua->argv[i][0];
-    } else if (Bstrcasecmp(ua->argv[i], "terminated")) {
-      *jobstatus = JS_Terminated;
-    } else if (Bstrcasecmp(ua->argv[i], "warnings")) {
-      *jobstatus = JS_Warnings;
-    } else if (Bstrcasecmp(ua->argv[i], "canceled")) {
-      *jobstatus = JS_Canceled;
-    } else if (Bstrcasecmp(ua->argv[i], "running")) {
-      *jobstatus = JS_Running;
-    } else if (Bstrcasecmp(ua->argv[i], "error")) {
-      *jobstatus = JS_ErrorTerminated;
-    } else if (Bstrcasecmp(ua->argv[i], "fatal")) {
-      *jobstatus = JS_FatalError;
-    } else {
-      /* invalid jobstatus */
+  int i;  // index of job status argument
+
+  if ((i = FindArgWithValue(ua, NT_("jobstatus"))) >= 0) {  // jobstatus exists
```

Comment:
Comments that do not add new information are unnecessary
```suggestion
  if ((i = FindArgWithValue(ua, NT_("jobstatus"))) >= 0) {
```